modyn/storage/internal/grpc/storage_grpc_servicer.py

```diff
@@ -72,7 +72,10 @@ def Get(self, request: GetRequest, context: grpc.ServicerContext) -> Iterable[Ge
                 return
 
             samples: list[Sample] = (
-                session.query(Sample).filter(Sample.sample_id.in_(request.keys)).order_by(Sample.file_id).all()
+                session.query(Sample)
+                .filter(and_(Sample.sample_id.in_(request.keys), Sample.dataset_id == dataset.dataset_id))
```

Comment:
does that fix #286?

Reply:
The training works and I haven't seen any side effects but I don't know if it fixes it completely